src/ballet/gossip/fd_gossip_pretty_print.c

```diff
@@ -0,0 +1,265 @@
+
+    FD_LOG_WARNING(( "  => LowestSlot" ));
+    FD_LOG_WARNING(( "     -----------" ));
+    FD_LOG_WARNING(( "  - crds_id: 0x%x", crds_id & 0xff ));
+    fd_gossip_pretty_print_signature( "  - signature", &(lowest_slot->hdr.signature) );
+    fd_gossip_pretty_print_pubkey( "  - from", &(lowest_slot->data.from) );
+    FD_LOG_WARNING(( "  - root: addr = %lu", lowest_slot->data.root ));
+    FD_LOG_WARNING(( "  - lowest: addr = %lu", lowest_slot->data.root ));
+    FOR_EACH_U64_IN_VECTOR( lowest_slot, lowest_slot->data.slots, value ) {
+      FD_LOG_WARNING(( "       - 0x%lx ", value ));
+    }
+    FD_LOG_WARNING(( "  - stash: obsolete" ));
+    FD_LOG_WARNING(( "  - wallclock: addr = %lu", lowest_slot->data.wallclock ));
+
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_lowest_slot_t );
+
+    break;
+  }
+
+  case FD_GOSSIP_CRDS_ID_SNAPSHOT_HASHES:
+  case FD_GOSSIP_CRDS_ID_ACCOUNT_HASHES:
+    if( crds_id==FD_GOSSIP_CRDS_ID_SNAPSHOT_HASHES ) {
+      FD_LOG_WARNING(( "  => SnapshotHashes" ));
+      sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_snapshot_hashes_t );
+    }
+    else {
+      FD_LOG_WARNING(( "  => AccountHashes" ));
+      sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_account_hashes_t );
+    }
+    break;
+
+  case FD_GOSSIP_CRDS_ID_EPOCH_SLOTS: 
+    FD_LOG_WARNING(( "pretty-printing for EpochSlots not yet implemented" ));
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_epoch_slots_t );
+    break;
+
+  case FD_GOSSIP_CRDS_ID_LEGACY_VERSION:
+    FD_LOG_WARNING(( "pretty-printing for EpochSlots not yet implemented" ));
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_legacy_version_t );
+    break;
+
+  case FD_GOSSIP_CRDS_ID_VERSION:
+    FD_LOG_WARNING(( "pretty-printing for Version not yet implemented" ));
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_version_t );
+    break;
+
+  case FD_GOSSIP_CRDS_ID_NODE_INSTANCE:
+    FD_LOG_WARNING(( "pretty-printing for NodeInstance not yet implemented" ));
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_node_instance_t );
+    break;
+
+  case FD_GOSSIP_CRDS_ID_DUPLICATE_SHRED:
+    FD_LOG_WARNING(( "pretty-printing for DuplicateShred not yet implemented" ));
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_duplicate_shred_t );
+    break;
+
+  case FD_GOSSIP_CRDS_ID_INCREMENTAL_SNAPSHOT_HASHES:
+    FD_LOG_WARNING(( "pretty-printing for IncrementalSnapshotHashes not yet implemented" ));
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_incremental_snapshot_hashes_t );
+    break;
+
+  case FD_GOSSIP_CRDS_ID_CONTACT_INFO:
+    FD_LOG_WARNING(( "pretty-printing for ContactInfo not yet implemented" ));
+    sz_to_advance_ptr = sizeof ( fd_gossip_crds_value_contact_info_t);
+    break;
+
+  default:
+    FD_LOG_WARNING(( "unknown CRDS value, %u", crds_id ));
+    break;
+  }
+
+  uchar *ptr = (uchar *)(*crds_hdr);
+  ptr += sz_to_advance_ptr;
```

Comment:
src/ballet/gossip/fd_gossip_pretty_print.c:138:10: error: variable 'ptr' set but not used [-Werror,-Wunused-but-set-variable]
  uchar *ptr = (uchar *)(*crds_hdr);
         ^